src/main/resources/interface-specification.yml

```diff
@@ -748,6 +798,42 @@ components:
             $ref: '#/components/schemas/Agreement'
       required:
         - agreements
+    Messages:
+      type: object
+      properties:
+        limit:
+          type: integer
+          format: int32
+        size:
+          type: integer
+          format: int32
+        nextId:
+          type: string
```

Comment:
thinking aloud: could this pagination be simplified?
If we just expect to receive another request to `/events`, with the last `eventId` that was in the previous response, could we remove `nextId` (and in the same way `limit` and `size`)?

Reply:
sure, it is the canonical approach and it's totally fine.
What I was thinking is that the two logics are somewhat overlapping. 
For example, this request
```
/events?eventId=10&limit=100&nextId=qwerty
```
is the same as
```
/events?eventId=110&limit=100
```
so why don't we use just one of them?

Reply:
- I agree to drop the `size` parameter, the caller of this api should not worry about the size of the message (this under-the-hood implementation should be masked)
- the `limit` in the response is a utility, personally nothing against that unless it confuses the user
- about the `nextId` my point was: is it really necessary in this specific endpoint?

I'll detail more my previous example:
### Request 1
```
/events?eventId=0&limit=3
```
### Response 1
```json
{
  "messages": [
     { "eventId": 1 },
     { "eventId": 2 },
     { "eventId": 3 },
  ]
}
```
### Request 2
```
/events?eventId=3&limit=100
```
So here we could take the last `eventId` of the response and use it to request further messages.
But it was just an idea, if you think it is not straightforward we can keep the `nextId`.

Reply:
I am fine with the proposal.

Reply:
ℹ️  as result of discussions with Marco and @P3trur0, it makes sense to include in the response the info of the next page, but it will be the last event id received instead of the next id yet to be received